common/concurrency/WorkerPool.h

```diff
@@ -14,6 +14,11 @@ class WorkerPool {
     using Task = std::function<void()>;
     static std::unique_ptr<WorkerPool> create(int size, spdlog::logger &logger);
     virtual void multiplexJob(std::string_view taskName, Task t) = 0;
```

Comment:
Can we document this one too, something like

"does not block after launching the workers, usually useful so that the main thread can have a consumer queue that streams the results of the worker threads as they produce them"

Reply:
Absolutely!